src/services/charmcraft/charmcraft.ts

```diff
@@ -115,13 +116,39 @@ class Charmcraft {
       throw new Error('Could not pull the docker image.');
     }
 
+    // strip any repository details from the image name, as this will confuse `docker image ls`
+    const resourceImageNameOnly = getImageName(resource_image);
+
+    const result = await getExecOutput(
```

Comment:
I would prefer to have this block in docker.ts, since it's a more docker specific operation. Otherwise, LGTM

Reply:
good point.  Moved that into the helper